File: 006.clj
(defn sum-of-squares [n]
  (apply + (map #(* % %) (range 1 (+ n 1)))))

(defn square-of-sum [n]
  (*
    (apply + (map #(+ %) (range 1 (+ n 1))))
    (apply + (map #(+ %) (range 1 (+ n 1))))))

(println (- (square-of-sum 100) (sum-of-squares 100)))

File: 006.hs
-- Find the difference between the sum of the squares of the 
-- first one hundred natural numbers and the square of the sum.

sumSquares = sum $ map (^2) [1..100]
squareSum = (^2) $ sum [1..100]

main = print $ squareSum - sumSquares

-- 25164150
